src/Engine/Input/InputManager.cpp

```
@@ -32,7 +32,13 @@ void InputManager::bindEvents() {
 
 
 void InputManager::tick() {
-	processKeyboardInput(Time::GetDeltaTime());
+    // NOTE: Instead of Time::GetDeltaTime, we use render-frame delta time for input-driven movement so that the camera movement is based on actual frame time instead of the physics thread's delta
+	static auto previousTime = std::chrono::high_resolution_clock::now();
+	auto currentTime = std::chrono::high_resolution_clock::now();
+	double frameDelta = std::chrono::duration<double>(currentTime - previousTime).count();
```

Clang (14.0.6)

[warning] variable 'frameDelta' is not initialized

(cppcoreguidelines-init-variables)